token.go

```diff
@@ -174,7 +175,10 @@ func (t *simpleTokenManager) getRetrieveNewTokenClosure(uri *url.URL) func(chan
 			req.Header.Set("updated_at_apid", time.Now().Format(time.RFC3339))
 		}
 
-		client := &http.Client{Timeout: httpTimeout}
+		client := &http.Client{
```

Comment:
Your code looks good.
In fact, we should reuse the same `http.Client` in this plugin or even whole apid. () Creating a new `http.Client` on the fly for getting a token is not correct. This is a separate task. I'll fix it after your fwd-proxy merged, to avoid conflicts.

Reply:
We cannot have a single client for apid, as each plugin can have different client characteristics.